var/spack/repos/builtin/packages/onednn/package.py

```diff
@@ -75,22 +77,35 @@ class Onednn(CMakePackage):
         "cpu_runtime",
         default=default_cpu_runtime,
         description="CPU threading runtime to use",
-        values=("omp", "tbb", "seq"),
+        values=(
+            conditional("none", when="@2.3:"),
+            "omp",
+            "tbb",
+            "seq",
+            conditional("threadpool", when="@1.4:"),
+            conditional("dpcpp", when="@2:"),
+            conditional("sycl", when="@2:"),
+        ),
         multi=False,
     )
     variant(
         "gpu_runtime",
         default="none",
         description="Runtime to use for GPU engines",
-        values=("ocl", "none"),
+        values=("ocl", "none", conditional("dpcpp", when="@2:"), conditional("sycl", when="@2:")),
         multi=False,
     )
+    variant(
+        "acl", default=False, description="Use Arm Compute Library", when="@1.7: target=aarch64:"
+    )
 
     # https://github.com/oneapi-src/oneDNN#requirements-for-building-from-source
+    depends_on("cmake@2.8.12:", when="@2.3:", type="build")
     depends_on("cmake@2.8.11:", type="build")
     depends_on("tbb@2017:", when="cpu_runtime=tbb")
     depends_on("llvm-openmp", when="%apple-clang cpu_runtime=omp")
     depends_on("opencl@1.2:", when="gpu_runtime=ocl")
```

Comment:
Also need to add a few dependencies for these new runtimes:
```python
depends_on("tbb", when="cpu_runtime=sycl")
depends_on("opencl@1.2:", when="gpu_runtime=sycl")
depends_on("openapi-level-zero", when="gpu_runtime=sycl")
```
SYCL also requires `%dpcpp` but there isn't an easy way to express that right now so we can prob just ignore it.

Reply:
Done.